### PR TITLE
test: add custom TLS tests

### PR DESCRIPTION
**Pull request by cpanato**
fixes: INST-184

### CODE REVIEW

**Comment by cpanato** [comment]
retest this please

File: tests/rspec/lib/tls_certs.rb [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  ].each do |cert_name, contents|
+    save_to_file(path, cert_name, contents)
+  end
+end
+
+def generate_api_certs(cluster_name, domain)
+  root_kube = certificate_authority('kube-ca', 'bootkube')
+
+  dns_main_node = if domain.include?('example.com')
+                    'cluster.example.com'
```
Comment (mxinden) [inline tests/rspec/lib/tls_certs.rb]: Is this differentiating bare metal and all other platforms? If so, could you add a comment?

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
@mxinden can you please take another look? thanks! :-)

File: modules/dns/gcp/records.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 resource "google_dns_record_set" "api-external" {
-  name         = "${var.cluster_name}.api.${var.base_domain}."
+  name         = "${var.cluster_name}-api.${var.base_domain}."
```
Comment (s-urbaniak) [inline modules/dns/gcp/records.tf]: /cc @enxebre

**Pull request merged**